File: 2022-2023/sourcecode/lab/lab11/RetrofitUrlBasicsJava/app/src/main/java/pl/udu/uwr/pum/retrofiturlbasicsjava/PlaceholderService.java
package pl.udu.uwr.pum.retrofiturlbasicsjava;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;

public interface PlaceholderService {

    @GET("posts/{id}/comments")
    Call<List<Comment>> getComments(@Path("id") int postId);

    @GET("comments")
    Call<List<Comment>> getCommentsFromQuery(@Query("postId") int postId);

    @GET("comments")
    Call<List<Comment>> getSortedComments(
            @Query("postId") int postId,
            @Query("_sort") String sort,
            @Query("_order") String order);

    @GET("comments")
    Call<List<Comment>> getNullableComments(
            @Query("postId") Integer postId,
            @Query("_sort") String sort,
            @Query("_order") String order);

    @GET("comments")
    Call<List<Comment>> getSortedCommentsFromPosts(
            @Query("postId") Integer[] postsId,
            @Query("_sort") String sort,
            @Query("_order") String order);

    @GET("comments")
    Call<List<Comment>> getComments(
            @QueryMap Map<String, String> param
    );

    @GET
    Call<List<Comment>> getComments(
            @Url String url
    );
}

File: 2022-2023/sourcecode/lab/lab11/RetrofitUrlBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/retrofiturlbasicskotlin/PlaceholderService.kt
package pl.udu.uwr.pum.retrofiturlbasicskotlin

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import retrofit2.http.QueryMap
import retrofit2.http.Url

interface PlaceholderService {
    @GET("posts/{id}/comments")
    fun getComments(@Path("id") postId: Int): Call<List<Comment>>

    @GET("comments")
    fun getCommentsFromQuery(@Query("postId") postId: Int): Call<List<Comment>>

    @GET("comments")
    fun getSortedComments(
        @Query("postId") postId: Int,
        @Query("_sort") sort: String,
        @Query("_order") order: String
    ): Call<List<Comment>>

    @GET("comments")
    fun getNullableComments(
        @Query("postId") postId: Int,
        @Query("_sort") sort: String,
        @Query("_order") order: String
    ): Call<List<Comment>>

    @GET("comments")
    fun getSortedCommentsFromPosts(
        @Query("postId") postsId: List<Int>,
        @Query("_sort") sort: String,
        @Query("_order") order: String
    ): Call<List<Comment>>

    @GET("comments")
    fun getComments(
        @QueryMap param: Map<String, String>
    ): Call<List<Comment>>

    @GET
    fun getComments(
        @Url url: String
    ): Call<List<Comment>>
}
